### PR TITLE
Final Polish translations

### PR DESCRIPTION
**Pull request by pauldim**
Final Polish translations. This PR is co-created with scr0llbaer9000. We can do the rest of the shortcomings later.

### CODE REVIEW

**Comment by scr0llbaer9000** [comment]
Did not come around for a final round of checks / fixes:

- All references to roles in `eggi_screenplays_adaptations.xml` must be fixed (wrong quotation marks messed up by translation tool) (had noticed that, but forgot).

- `database_programmes.xml` has a `[2|Nicka]a` that must be fixed to `${.self:"cast":2:"nickname"}a` (validator kinda finds that by coincidence)

- variable options count mismatches with the other languages in `database_scripts.xml` in lines 394, 939, 970, 980, 1021

- variable options count mismatch in `scr0llbaer_news.xml in lines` 1336 f (forgot some `|` characters at the end)

- variable options count mismatch in `scr0llbaer_screenplays_adaptations.xml` in lines 6292, 7356, 7694, 9517

- property name got errouneously tanslated in `scr0llbaer_screenplays_adaptations.xml` line 1694 (must be `"firstname"` instead of `"imię"`)

That's all I have for now.

btw validator doesn't update in my eclipse @nittka , is that problem on your side? So it's still a bit hard to wade through everything.

**Comment by nittka** [comment]
I will review the changes and give additional feedback. After the merge, my suggetion would be for pauldim to create a backup-branch of his current master (if wanted) and do a hard-reset + force-push on the official master (cannot be done in the browser, but must be done from a client). In the standard case, the fork's master should be kept in synch by pulling/fast forward merges rather than merge commits. Contributions/on changes should be done on a branch and not the master.

@scr0llbaer9000 : There are still some PRs open that have impact on the editor. I wanted to know if they are merged or rejected before I release a new editor version.

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 1]
```diff
@@ -1532,14 +1691,17 @@
 			<title>
 				<de>Knirps</de>
 				<en>${.self:"role":1:"firstname"}</en>
+				<pl>${.self:"role":1:"imię"}</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:1694]: parameter firstname - not translated

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 2]
```diff
@@ -5560,30 +6270,36 @@
...
 			<variables>
 				<title_original_optional>
 					<de>Dead Winter Dead|Tot Winter Tot|Tot Winter Tod|Toter Wintertod|Tote Winterstarre|Toter Winter in Sarjevo|${dead} ${winter} in ${sarajevo}</de>
 					<en>Dead Winter Dead|Dead Winter Dead|Dead Winter Death|Dead Winter Death|Dead Winter Dead|Dead Winter in Sarajevo|${dead} ${winter} in ${sarajevo}</en>
+					<pl>Martwa zima nie żyje|Martwa zima martwa|Martwa zima śmierci|Martwa zima śmierci|Martwa zima martwa|Martwa zima w Sarajewie|${dead} ${winter} w ${sarajevo}</pl>
 				</title_original_optional>
 				<dead>
 					<de>Toter|Karger|Kalter|Trostloser|Trister|Rauer</de>
 					<en>Dead|Desolate|Cold|Dismal|Dreary|Bleak</en>
+					<pl>Martwy|Opuszczony|Zimny|Ponury|Mroczny</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:6292]: 5 alternatives here, 6 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 3]
```diff
@@ -6476,47 +7337,58 @@
...
 				</title_spoofed>
 				<title_original_optional>
 					<de>Maniac Mansion|Der ${adj_o} ${house_m_o}|Die ${adj_o} ${house_f_o}|Das ${adj} ${house_n_o}</de>
 					<en>Maniac Mansion|The ${adj_o} ${house_m_o}|The ${adj_o} ${house_f_o}|The ${adj} ${house_n_o}</en>
+					<pl>Maniac Mansion|Maniac Mansion -${adj_o} ${house_m_o}|Maniac Mansion - ${adj_o} ${house_f_o}|Maniac Mansion - ${adj_o} ${house_n_o}</pl>
 				</title_original_optional>
 				<adj>
 					<de>Wahnwitzige|Verrückte|Irrwitzige|Durchgeknallte|Geisteskranke|Irrsinnige|Aberwitzige|Manische|Wahnsinnige|Absurde|Groteske|Bizarre|Abgedrehte</de>
 					<en>Lunatic|Crazy|Mad|Frenzied|Frantic|Insane|Ludicrous|Manic|Maniacal|Absurd|Grotesque|Bizarre|Wacky</en>
+					<pl>Szalona|Wściekła|Oszalała|Rozgorączkowana|Obłąkana|Śmieszna|Maniacka|Maniakalna|Absurdalna|Groteskowa|Dziwaczna|Odlotowa</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:7356]: 12 alternatives here, 13 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 4]
```diff
@@ -6770,39 +7670,48 @@
...
 			<variables>
 				<secret_title>
 					<de>Mysterium|Rätsel|Geheimnis</de>
 					<en>Mystery|Riddle|Enigma</en>
+					<pl>Tajemnica|Zagadka|Enigma</pl>
 				</secret_title>
 				<secret_title_original>
 					<de>Geheimnis|Mysterium|Rätsel|Geheimnis|Geheimnis</de>
 					<en>Secret|Mystery|Riddle|Enigma|Secret</en>
+					<pl>Tajemnica|Zagadka|Mit|Sekret</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:7694]: 4 alternatives here, 5 in the other languages

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 5]
```diff
@@ -7313,147 +8312,179 @@ The project, which was cancelled by a major producer, is now free to be realized
...
 			<variables>
 				<title_spoofed>
 					<de>Die ${space}${adventures} des ${captain} ${.self:"role":1:"nickname"}|Die ${galactic} ${adventures_c} des ${captain} ${.self:"role":1:"nickname"}</de>
 					<en>The ${space} ${adventures} of ${captain} ${.self:"role":1:"nickname"}|The ${galactic} ${adventures_c} of ${captain} ${.self:"role":1:"nickname"}</en>
+					<pl>${space} ${adventures} ${captain} Keena Billego Blaze'a|${galactic} ${adventures_c} ${captain} Keena Billego Blaze'a</pl>
 				</title_spoofed>
 				<title_original_optional>
 					<de>${captain} ${.self:"role":1:"nickname"}|Die ${space}${adventures} des ${captain} ${.self:"role":1:"nickname"}|Die ${galactic} ${adventures_c} des ${captain} ${.self:"role":1:"nickname"}</de>
 					<en>${captain} ${.self:"role":1:"nickname"}|The ${space} ${adventures} of ${captain} ${.self:"role":1:"nickname"}|The ${galactic} ${adventures_c} of ${captain} ${.self:"role":1:"nickname"}</en>
+					<pl>Keen Billy Blaze|${space} ${adventures} ${captain} Keena Billego Blaze'a|${galactic} ${adventures_c} ${captain} Keena Billego Blaze'a</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:8326]: no role references here?

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml [hunk 6]
```diff
@@ -8362,164 +9498,199 @@ The project, which was cancelled by a major producer, is now free to be realized
...
+				<pl>Nowe wydanie popularnego czeskiego serialu kukiełkowego o ojcu ${.self:"role":1:"fullname"}, synku ${.self:"role":2:"firstname"}, przyjaciółce ze szkoły ${.self:"role":4:"firstname"}, babci pani ${.self:"role":3:"firstname"} i psie ${.self:"role":5:"firstname"}. Ostrożnie zmodernizowane, aby zachować urok.</pl>
 			</description>
 			<variables>
 				<new>
 					<de>Neuen|Neuartigen|Weiteren|Frischen|Ungeahnten|Nachfolgenden|Unversuchten|Ursprünglichen|Neuerlichen|Wiederbelebten|Erneuten|Neugeborenen|Wiedererstarkten|Reanimierten|Auferstandenen</de>
 					<en>New|Novel|Further|Fresh|Unprecedented|Successive|Untried|Pristine|Refreshed|Revived|Renewed|Reborn|Invigorated|Reanimated|Resurrected</en>
+					<pl>Nowe|Nowe|Dalsze|Świeże|Bezprecedensowe|Kolejne|Niewypróbowane|Nieskazitelne|Odświeżone|Ożywione|Odnowione|Odrodzone|Ożywione|Ożywione</pl>
```
Comment (nittka) [inline res/database/Default/user/scr0llbaer_screenplays_adaptations.xml:9517]: 14 alternatives here, 15 in the other languages

**Review by nittka — CHANGES_REQUESTED** [review]
I still have to go through the changes in the main programmes file (old cast reference in line 84566 [2|Nick...).
Please fix the formal issues first. Only after that is done, it makes sense to run the test script that tries to find unresolvable references/variables.

**Comment by nittka** [comment]
In programmes.xml very many original titles for French/Italien etc. have been removed. (I compared the files from master and this PR with all Polish tags removed - ideally they should be largely identical).

File: res/database/Default/user/eggi_screenplays_adaptations.xml [hunk 7]
```diff
@@ -10,10 +10,12 @@
 			<title_original>
 				<de>Wellness</de>
 				<en>Wellness</en>
+				<pl>Wellness</pl>
 			</title_original>
 			<description>
 				<de>${.self:"role":1:"firstname"} und ${.self:"role":2:"firstname"}, ungleiches Paar in der Kunstszene Chicagos, kämpfen gegen Eheprobleme. Zwischen Achtsamkeitsseminaren und Immobilienträumen müssen sie ihre Vergangenheit bewältigen, um ihre Liebe zu retten.</de>
 				<en>${.self:"role":1:"firstname"} and ${.self:"role":2:"firstname"}, a mismatched couple in Chicago's art scene, are battling marital problems. Between mindfulness seminars and real estate dreams, they have to come to terms with their past in order to save their love.</en>
+				<pl>${.self: „role”:1: „firstname”} i ${.self: „role”:2: „firstname”}, niedopasowana para z chicagowskiej sceny artystycznej, zmaga się z problemami małżeńskimi. Pomiędzy seminariami i marzeniami o nieruchomościach, muszą pogodzić się ze swoją przeszłością, aby uratować swoją miłość.</pl>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/user/eggi_screenplays_adaptations.xml]: Done

File: res/database/Default/database_news.xml [hunk 8]
```diff
@@ -8261,12 +8564,15 @@ W tym roku ponownie spodziewane są demonstracje.</pl>
...
+					<de>Der ${age}-jährige Schüler;er;seinem;${.persongenerator:"fullname":"de":"male"};|
+						Die ${age}-jährige Schülerin;sie;ihrem;${.persongenerator:"fullname":"de":"female"};
 					</de>
-					<en>The ${age} year old schoolboy;he;his;${.persongenerator:"fullname":"en":"male"}|
-						The ${age} year old schoolgirl;she;her;${.persongenerator:"fullname":"en":"female"}
+					<en>The ${age} year old schoolboy;he;his;${.persongenerator:"fullname":"en":"male"};|
+						The ${age} year old schoolgirl;she;her;${.persongenerator:"fullname":"en":"female"};
 					</en>
+					<pl>${age} letni uczeń;on;jego;${.persongenerator:"fullname":"pl":"male"};Złamał|
+						${age} letnia uczenica;ona;jej;${.persongenerator:"fullname":"pl":"female";Złamała}
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/database_news.xml:8574]: Corrected

File: res/database/Default/database_news.xml [hunk 9]
```diff
@@ -7842,109 +8067,129 @@ W tym roku ponownie spodziewane są demonstracje.</pl>
...
 		<news guid="7cb237a1-f3cd-49fb-a553-e31bed1a882c" type="2" thread_guid="96e6549c-19a1-4ede-a125-a665c1ae87de" creator="" created_by="-">
 			<title>
 				<de>${armwrestling_${i}}-Game großer Erfolg</de>
 				<en>${armwrestling_${i}} game big success</en>
+				<pl>Wielki sukces gry o ${armwrestling_${i}}</pl>
 			</title>
 			<description>
 				<de>Nachdem nach der Weltmeisterschaft ein kleiner Hype entstanden ist, führt das Game „${armwrestling_${i}} Challenge“ von ${artistlastname}soft derzeit die internationalen Verkaufscharts an.</de>
 				<en>After the minor hype following the World Championship, the game "${armwrestling_${i}} Challenge" by ${artistlastname}soft is currently topping the international sales charts.</en>
+				<pl>Po szumie po Mistrzostwach Świata, gra „Wyzwania w ${armwrestling_${i}}“ autorstwa ${artistlastlastname}soft znajduje się obecnie na szczycie międzynarodowych list sprzedaży.</pl>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/database_news.xml:8114]: There is so much to look at and so many mistakes. Corrected.

File: res/database/Default/database_scripts.xml [hunk 10]
```diff
@@ -636,81 +712,100 @@
...
 						</description_4>
 						<description_5>
 							<de>wollen in der Nachmittagsvorstellung unterhalten werden, obwohl sonst kaum Publikum da ist. Die ${who} machen die tote Hose bunt.</de>
 							<en>want to be entertained in the afternoon performance, even though there is hardly any other audience. The ${who} color the dullness.</en>
+							<pl>chcą być zabawiane podczas popołudniowego występu, mimo że nie mają prawie żadnej innej publiczności. ${.ucfirst:who} barwią nudę.</pl>
 						</description_5>
 						<description_6>
 							<de>sind da zur Beschwerde, weil sie mit der Behandlung ihrer Opas nicht einverstanden sind. Die ${who} mediieren gewaltfrei.</de>
 							<en>are there to complain because they don't agree with the treatment of their grandfathers. The ${who} mediate non-violently.</en>
+							<pl>są tam, żeby narzekać, bo nie zgadzają się z traktowaniem swoich dziadków. ${.ufirst:who} pośredniczą bez przemocy.</pl>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/database_scripts.xml:745]: Done

File: res/database/Default/user/scr0llbaer_screenplays_adaptations.xml (hunk 5 above)
Comment (pauldim) [reply res/database/Default/user/scr0llbaer_screenplays_adaptations.xml]: Grammar difficulties :)

File: res/database/Default/user/ariksan.xml [hunk 11]
```diff
@@ -4,16 +4,20 @@
 	<scripttemplates>
 		<scripttemplate product="2" licence_type="3" guid="script-ariksan-mkw" creator="9621" comment="inspired by German realtor docu soaps">
 			<title>
-				<all>${title} in ${randomcity}</all>
+				<de>${title} in ${randomcity}</de>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/user/ariksan.xml:7]: <pl>${title} w ${randomcity}</pl> - There must be a Polish version

File: res/database/Default/database_ads.xml [hunk 12]
```diff
@@ -233,19 +259,21 @@ You will really have to chew.</en>
...
 				<en>The game without boundaries. The ${brand_ps}5 is the successor to the ${brand_ps}4 made by ${brand_sony}.</en>
+				<pl>Gra bez granic. ${brand_ps}5 jest następcą ${brand_ps}4 wyprodukowanym przez ${brand_sony}.</pl>
 			</description>
 			<conditions min_audience="9.38" min_image="18" target_group="2" pro_pressure_groups="0" contra_pressure_groups="0" />
 			<data quality="15" repetitions="3" duration="3" profit="335" penalty="550" infomercial_profit="33" fix_infomercial_profit="1" />
 			<availability year_range_from="2020" />
 		</ad>
 		<ad guid="3bf32cbc-3322-4f3b-851b-88b964cb990d" comment="https://en.wikipedia.org/wiki/MediaMarkt | https://en.wikipedia.org/wiki/Best_Buy | TODO: Maybe separate these ads after all when ad images feature arrives">
 			<title>
+				<all>${brand_bestbuy}</all>
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/database_ads.xml:270]: Media Markt in Poland. Great , that we have this functionality of original ads. I just noticed it.

**Review by nittka — CHANGES_REQUESTED** [review]
In case you are interested where Polish translations (title) are still missing (probably last minute additions in one of the recent commits):
* scr0llbaer.xml line 1271 and following: d26ee4f9-1b61-4b7a-926a-e462256b127b and episodes
* scr0llbaer.xml line 1354: d664f243-15d6-4645-886e-27a3e8cf8533
* database_ads.xml line: 360b9ee1-9e2e-469d-b61e-fa8a69cbf560 (set brand explicitly)
* database_programmes.xml line 7622: 216d7e50-2893-4a2e-973a-4f55a3ab81b0
* database_programmes.xml line 53814: 6543de67-43ce-4a32-ab3f-c7c1dbbae34a

File: res/database/Default/user/ariksan.xml [hunk 13]
```diff
@@ -344,342 +387,416 @@
 				<animal2>
 					<de>Nashorn|Flusspferd|Pferd|Stachelschwein|Schaf|Huhn|Kamel|Walross|Krokodil</de>
 					<en>rhino|hippo|horse|porcupine|sheep|chicken|camel|walrus|crocodile</en> <!-- should not be added: lion, cobra (already appear in episodes) -->
+					<pl>nosorożec|hipopotam|koń|jeżozwierz|niedźwiedź|kurczak|wielbłąd|mors|krokodyl</pl>
 				</animal2>
 				<adjective>
 					<de>witzige|aufgeweckte|gescheite|kluge|lustige</de>
 					<en>Witty|Sharp|Clever|Smart|Funny</en>
+					<!-- In PL  <adjective_m> or <adjective_w> -->
```
[the remark this answers is not in the document]
Comment (pauldim) [reply res/database/Default/user/ariksan.xml]: Like this:
<adjective>
					<de>witzige|aufgeweckte|gescheite|kluge|lustige</de>
					<en>Witty|Sharp|Clever|Smart|Funny</en>
					<pl>Dowcipn|Ostr|Sprytn|Inteligentn|Zabawn</pl>
				</adjective>
				<adjective_m>
					<pl>${adjective}y</pl>
				</adjective_m>
				<adjective_w>
					<pl>${adjective}a</pl>
				</adjective_w>
Corrected

**Comment by nittka** [comment]
I started checking with the test script...
* database_programmes line 112010: you fixed the reference, but the square brackets are still present `[` and `]` should be removed
* ariksan line 511: variable name ${animal2} was translated
* database_scripts line 1104: variable ${who_${i}} should be replaced by ${who}
* scr0llbar_news line 1362: the number of entries within the csv tvtowermeta_csv does not match. It looks like one pair of translations is missing (8 entries instead of 10). Also, are the white spaces after the semicolon intended? They will be present in the resulting text.
* database_news line 3981: again too few csv elements; I guess, the comma after Rio de Janeiro should be a semicolon. Note also that in the alternatives of this entry, there is always a white space in the second entry except for London (first alternative)

**Comment by nittka** [comment]
database_news line 7000: one csv entry missing - i guess the comma after heavy metalovy should be a semicolon

**Comment by pauldim** [comment]
`<scripttemplate product="1" licence_type="1" guid="443fa70e-450c-4085-a4de-c41b23e295ef" creator="9551" comment="Description largely by ChatGPT">
			<title>
				<de>Sekretärinnen-Bericht 3. Teil - Stenografie, Streiche &amp; Sinnlichkeit</de>
				<en>Secretaries Report Part 3 - Shorthand, Stunts &amp; Sensuality</en>
				<pl>Raport sekretarek, Część 3 - Skróty, akrobacje i zmysłowość</pl>
			</title>
			<description>
				<de>Jetzt wird's noch bunter: ${.self:"role":1:"fullname"} wird zur Meisterin der Stenografie und anderer Quickies, Fax und Fon klingeln im Rhythmus der Aktion und ${.self:"role":2:"fullname"}s klappernder Schreibmaschine... Inmitten biederer Bürokratie brodelt feurige Fantasie.</de>
				<en>Now it all gets extra sparkly: ${.self:"role":1:"fullname"} becomes a master of shorthand and other quickies, fax and phone ring to the rhythm of the action and ${.self:"role":2:"fullname"}'s clattering typewriter.**,,** In the midst of bland bureaucracy, fiery fantasies are fizzing.</en>`

There should be two commas in the English version?

**Comment by nittka** [comment]
No, I think the two commas should be dots - as in German. typewriter... In the midst

**Comment by pauldim** [comment]
I think everything has been corrected. I don't think I missed anything.

**Comment by scr0llbaer9000** [comment]
> After the merge, my suggetion would be for pauldim to create a backup-branch of his current master (if wanted) and do a hard-reset + force-push

...or just delete the TVTower fork and fork TVTower again. And then create a *branch* for your own experiments.

**Comment by nittka** [comment]
Test script and editor validation look fine. I'll merge tomorrow if I don't hear any objections.

**Comment by scr0llbaer9000** [comment]
The texts for the new blockbuster achievements in [database_achievements.xml](https://github.com/TVTower/TVTower/blob/master/res/database/Default/database_achievements.xml) are still untranslated.

Only very few lines that could be done in this PR, then that will be all as far as I can see.

**Comment by pauldim** [comment]
> @pauldim : please let me know, if you want to address the blockbuster achievement.

I didn't notice that comment. I've already added those translations.

**Comment by nittka** [comment]
Almost there. When working on improvements on the editor validation two subtle issues came up that the current validation and the test script missed (note that this script cannot guarantee 100% coverage of all possible alternatives, so even in combination editor and test script are not guaranteed to find all issues).

@scr0llbaer9000 and @pauldim please let me know if you want to correct those issues in this PR or separately (Polish translation is affected for one of them)

* trivial issue: scr0llbaer_news line 416 (733f6d6a-4021-45c0-8ace-7768f9a922f0) - in German the last alternative ends with pipe (|), i.e. there is another unintended empty alternative; remove the pipe
* less trivial: scr0llbar_news lines 630 and following (8ac54969-a412-4c23-98c1-a3b2f449605d) - the csv data alternatives 5 (all languages) and 6 (en and pl) of the variable what_csv are not consistent with the rest. All other entries have 3 data points (2x semicolon) but 5 and 6 have only 2 data points (1x semicolon). But the third datapoint is never referenced (...what_csv:2 cannot be found), so the texts there will never be used. Note that there is no support for escaping at the moment.

**Comment by scr0llbaer9000** [comment]
> trivial issue

not so trivial? or why didn't the validator find it before?

> scr0llbar_news lines 630 and following

ah yeah, these are "real" semicolons. Was wondering there might be instances of those. Originally used them because of casing. Can change to full stops now.

I'll fix these when you merge.

**Comment by nittka** [comment]
> or why didn't the validator find it before?

The bug simply never surfaced. If TVTower determines the used alternative based on English (no idea), the additional German one will never cause errors. If the 100 interations of the test script never chooses the last German one... With the validation working for the rest of the cases, there was no reason to check the code.

New editor version is available.

**Pull request merged**